### ISSUE TITLE
Show binding redirects for strongly signed assemblies

### ISSUE THREAD

**Issue by dotnetjunkie** [praeclarum/FuGetGallery#29]
Binding redirects are a constant pain when dealing with strongly signed assemblies, especially when they are references a lot, such as Newtonsoft.Json.

To easy this paint, it would be really nice if Fuget could show a binding redirect example for strongly signed assemblies.

For instance, this is an example that you could show when viewing the Newtonsoft.Json package:

``` xml
<runtime>
  <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
    <dependentAssembly>
	  <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" 
            culture="neutral" />
	  <bindingRedirect oldVersion="0.0.0.0-11.0.0.0" newVersion="11.0.0.0" />
    </dependentAssembly>
  </assemblyBinding>
</runtime>
```

Only the package version and the `publicKeyToken` have to be injected. I imagine this to be the template:

``` xml
<runtime>
  <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
    <dependentAssembly>
	  <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="{publicKeyToken}" 
            culture="neutral" />
	  <bindingRedirect oldVersion="0.0.0.0-{assemblyVersion}" newVersion="{assemlyVersion}" />
    </dependentAssembly>
  </assemblyBinding>
</runtime>
```
To be able to achieve this, two things need to be extracted from the dll itself:

- The public key token
- The assembly version, which can be something different than the NuGet package version.

The public key toekn can be received at runtime as follows:

``` c#
Assembly assembly = Assembly.ReflectionOnlyLoadFrom("path");

var publicKeyToken = 
    assembly.FullName.Split(new[] { ", " }, StringSplitOptions.None)
    .Where(p => p.StartsWith("PublicKeyToken="))
    .Select(p => p.Substring("PublicKeyToken=".Length))
    .FirstOrDefault();
```
If the `publicKeyToken` variable is `null`, it means the assembly is not signed and no binding redirect is required.

**Comment by stakx** [comment]
@dotnetjunkie, just FYI, your template assumes that assembly versions are steadily increasing for each new release of a project. This may be the case for most "well-behaved" projects, but there are cases where this rule won't hold.

For example, [Castle Core](https://github.com/castleproject/Core) regressed its assembly version from 4.1... or 4.2... (can't remember precisely) back to 4.0.0.0. Your template would then suggest a binding redirect for 0.0.0.0-4.0.0.0 -> 4.0.0.0, when the redirect recommended by the maintainers themselves is more like 0.0.0.0-4.999.999.999 -> 4.0.0.0.

TL;DR: Your suggestion might work well for most projects, just be aware that it cannot cover some special situations, where it would possibly make a bad recommendation.
